<launch>
<!-- Launch file for Livox Horizon LiDAR -->

	<arg name="rviz" default="true" />

	<rosparam command="load" file="$(find fast_lio_localization)/config/horizon.yaml" />

	<param name="feature_extract_enable" type="bool" value="0"/>
	<param name="point_filter_num" type="int" value="3"/>
	<param name="dense_publish_enable" type="bool" value="1" />
	<param name="filter_size_surf" type="double" value="1" />
	<param name="filter_size_map" type="double" value="1" />
	<param name="cube_side_length" type="double" value="1000" />
	<param name="runtime_pos_log_enable" type="bool" value="0" />
	<param name="pcd_save_enable" type="bool" value="0" />
    <node pkg="fast_lio_localization" type="fastlio_mapping" name="laserMapping" output="screen"  />


	<param name="max_len" type="int" value="20000"/>



	<!-- loalization-->
	<node pkg="fast_lio_localization" type="test_localization.py" name="global_localization" output="screen" />

	<param name="map_path" type="string" value="/home/wheeltec/wjd/fastlio/src/FAST_LIO/PCD" />

	<!-- transform  fusion-->
	<node pkg="fast_lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen" />


	<group if="$(arg rviz)">
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization.rviz" />
	</group>

</launch>